sentry-opentelemetry/lib/sentry/opentelemetry/span_processor.rb

```diff
@@ -0,0 +1,159 @@
+require 'singleton'
+
+module Sentry
+  module OpenTelemetry
+    TraceData = Struct.new(:trace_id, :span_id, :parent_span_id, :parent_sampled, :baggage)
+
+    class SpanProcessor < ::OpenTelemetry::SDK::Trace::SpanProcessor
+      include Singleton
+
+      SEMANTIC_CONVENTIONS = ::OpenTelemetry::SemanticConventions::Trace
+
+      # The mapping from otel span ids to sentry spans
+      # @return [Hash]
+      attr_reader :span_map
+
+      def initialize
+        @span_map = {}
```

Comment:
I think we may store `configuration.logger` and `configuration.instrumenter` here so we don't need to perform configuration lookup repeatedly.

Reply:
* if we do this, the precedence order of initialization comes back because you need access to the instantiated `Sentry.configuration` object 
* eventually we might do all the otel config automatically ourselves and then we can fix all precedence problems in our setup, but we don't want to make assumptions on user setups right now

for now, I will just kill the logger statements since they were mostly for development and keep the other stuff as is.